source/unified-test-format/unified-test-format.rst

```diff
@@ -2492,7 +2560,224 @@ An example of this operation follows::
         client: *client0
         connections: 1
 
+runOnThread
+~~~~~~~~~~~
+
+The "runOnThread" operation instructs the test runner to schedule an operation
+to be run on a given thread. runOnThread MUST NOT wait for the scheduled
+operation to complete. If any of the operation's test assertions fail, the
+entire test case MUST fail as well.
+
+The following arguments are supported:
+
+- ``thread``: Required string. Thread entity on which this operation should be
+  scheduled.
+
+- ``operation``: Required `operation`_ object. The operation to schedule on the
+  thread. This object must be a valid operation as described in `Entity Test
+  Operations`_.
+
+An example of this operation follows::
+
+     - name: runOnThread
+       object: testRunner
+       arguments:
+         thread: *thread0
+         operation:
+           name: insertOne
+           object: *collection0
+           arguments:
+             document: { _id: 2 }
+           expectResult:
+             $$unsetOrMatches:
+               insertedId: { $$unsetOrMatches: 2 }
+
+
+waitForThread
```

Comment:
`waitForPrimaryChange` and `wait` both support a `timeoutMS` option. Is there any reason to omit a timeout option (or the concept of a default timeout) for  `waitForThread` and `waitForEvent`? I would think either of these could also leave the test runner in a blocked state if there's no directive to abort and fail after some point.

Reply:
This `waitForThread` operation also omitted one in the original SDAM integration tests, though it doesn't state a reason why. The original `waitForEvent` operation did have a default timeout of 10 seconds and was accidentally omitted. For consistency, I'll just carry over the default timeout of 10 seconds for both of these operations.